### PR TITLE
Improve cluster address validation

### PR DESCRIPTION
**Pull request by freeekanayaka**

### CODE REVIEW

File: lxd/main_init_interactive.go [hunk 1]
```diff
@@ -116,8 +116,17 @@ func (c *cmdInit) askClustering(config *cmdInitData, d lxd.InstanceServer) error
...
+		for {
+			serverAddress = util.CanonicalNetworkAddress(cli.AskString(
+				fmt.Sprintf("What IP address or DNS name should be used to reach this node? [default=%s]: ", address), address, nil))
+			host, _, _ := net.SplitHostPort(serverAddress)
+			if shared.StringInSlice(host, []string{"", "[::]", "0.0.0.0"}) {
+				fmt.Printf("Invalid IP address or DNS name\n")
+				continue
+			}
+			break
+		}
```
Comment (stgraber) [inline lxd/main_init_interactive.go]: Wouldn't this be simpler as a validator function passed as the last argument to `AskString`?

**Comment by stgraber** [comment]
Using `net.IP` and the `IsGlobalUnicast` function is a better bet I think, it will also prevent you from setting a loopback address for example.

**Comment by freeekanayaka** [comment]
> Using `net.IP` and the `IsGlobalUnicast` function is a better bet I think, it will also prevent you from setting a loopback address for example.

I think loopback addresses should be supported tho. I use them a lot when testing locally.

File: lxd/main_init_interactive.go (hunk 1 above)
Comment (freeekanayaka) [reply lxd/main_init_interactive.go]: Good call. Fixed.

**Comment by stgraber** [comment]
Ah yeah, I guess loopback is useful for that

**Pull request merged**